src/case_encoder.h

```diff
@@ -72,34 +75,53 @@ class UpperCaseEncoder : public CaseEncoder {
   UpperCaseEncoder(bool removeExtraWhiteSpace)
   : removeExtraWhiteSpace_(removeExtraWhiteSpace) {}
 
-  std::pair<absl::string_view, int> normalizePrefix(absl::string_view input) {
+  std::pair<absl::string_view, int> normalizePrefix(absl::string_view orig_input) {
+    if((dump_buffer_from_ >= 0) && (dump_buffer_from_ < buffer_queue_.size())) {
```

Comment:
I think this "if" statement deserves a short comment explaining what is handled here.